ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,296 @@
+        'Instance {0} removed from Load Balancer {1}.'
+        .format(instance_id, elb_name))
+    _remove_instance_from_load_balancer_list_in_properties(instance_id)
+
+
+@operation
+def add_instance_to_load_balancer(**_):
+
+    ctx.logger.debug('Attempting to add_instance_to_load_balancer.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
+        lb = elb_client.get_all_load_balancers(
+            load_balancer_names=[elb_name])[0]
+        lb.register_instances(instance_id)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+            raise NonRecoverableError('Instance not added to Load Balancer. '
+                                      '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} added to Load Balancer {1}.'
+        .format(instance_id, elb_name))
+
+    _add_instance_to_load_balancer_list_in_properties(instance_id)
+
+
+def _add_health_check_to_load_balancer(load_balancer, health_check):
+
+    hc = _create_health_check(health_check)
+
+    try:
+        load_balancer.configure_health_check(hc)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Health check not added to Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Health check added to Load Balancer {0}.'
+        .format(load_balancer.name))
+
+
+@operation
+def use_external_elb(**_):
+
+    utils.get_external_resource_id_or_raise(
+        'elb_name', ctx.target.instance)
+
+    _verify_load_balancer_exists()
+
+
+@operation
+def delete_load_balancer(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        raise NonRecoverableError('Will not delete external Load Balancer. ')
+
+    ctx.logger.debug('Attempting to delete Load Balancer.')
+
+    elb_name = ctx.node.properties['elb_name']
+
+    elb_client = connection.ELBConnectionClient().client()
+
+    try:
+        lb = elb_client.get_all_load_balancers(
+            load_balancer_names=[elb_name])[0]
+        lb.delete()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer {0} not deleted. '
+                                  .format(str(e)))
+
+    ctx.logger.info(
+        'Load Balancer {0} deleted. '
+        .format(elb_name))
+    if 'elb_name' in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return True
+
+
+def _create_health_check(properties):
```

Comment:
I don't like the word "properties" here, because it is confusing. Use something more descriptive like "user_health_check"